src/components/content/document/patient-documents.tsx

```diff
@@ -51,11 +61,46 @@ function PatientDocumentsComponent({ className, enableFQS = false }: PatientDocu
         emptyMessage="There are no documents available."
         columns={patientDocumentColumns(featureFlags?.enableViewFhirButton)}
         onRowClick={openDetails}
+        rowActions={rowActions}
       />
     </div>
   );
 }
 
+type ExtraRowActionProps = {
+  onAddToRecord?: (document: DocumentModel, binary: fhir4.Binary) => void;
+};
+
+const getRowActions =
+  ({ onAddToRecord }: ExtraRowActionProps) =>
+  (props: RowActionsProps<DocumentModel>) =>
+    <RowActions {...props} onAddToRecord={onAddToRecord} />;
+
+type RowActionsProps2 = RowActionsProps<DocumentModel> & ExtraRowActionProps;
+
+const RowActions = ({ record, onAddToRecord }: RowActionsProps2) => {
+  const { t } = useBaseTranslations();
+  const { getRequestContext } = useCTW();
+  const { binaryID } = record;
+
+  return onAddToRecord && binaryID ? (
+    <div className="ctw-flex ctw-space-x-2">
+      <button
+        type="button"
+        className="ctw-btn-primary ctw-ml-1 ctw-capitalize"
+        data-zus-telemetry-click="Add to record"
+        data-testid="add-to-record"
+        onClick={async () => {
+          const binary = await getBinaryDocument(await getRequestContext(), binaryID);
+          onAddToRecord(record, binary);
+        }}
+      >
+        {t("resourceTable.add")}
+      </button>
+    </div>
+  ) : null;
```

Comment:
If we take the route above we should add a useState variable to this component like
```
const [actionError, setActionError] = useState<string>();
```
and down here we could catch any error and set it to `actionError`
```suggestion
  return rowAction && binaryID ? (
    <div className="ctw-flex ctw-space-x-2">
      <button
        type="button"
        className={cx(rowAction.className, "ctw-btn-primary ctw-ml-1 ctw-capitalize")}
        data-zus-telemetry-click="Document custom row action"
        data-testid="custom-row-action"
        onClick={async () => {
          try {
            const binary = await getBinaryDocument(await getRequestContext(), binaryID);
            rowAction.onAction(record, binary);
          } catch (err) {
            setActionError(err.message ?? "unknown error")
          }
        }}
      >
        {rowAction.text}
      </button>
    </div>
  ) : null;
```